pkg/sessions/redis/client.go

```diff
@@ -12,6 +12,10 @@ type Client interface {
 	Get(ctx context.Context, key string) ([]byte, error)
 	Set(ctx context.Context, key string, value []byte, expiration time.Duration) error
 	Del(ctx context.Context, key string) error
+
+	// close is only used in tests to ensure that client connections are terminated
+	// after each test is finished
+	close() error
```

Comment:
Any advantage to just doing these in the tests themselves directly rather than adding to the interface and implementation just for tests sake?

`redisStore.Client.(*client).Close()`

And for cluster:

`redisStore.Client.(*clusterClient).Close()`

Reply:
I'm happy either way to be honest, it's not uncommon for things like this to be implemented for the tests sake within Go I don't think, but I could just do as you've suggested.

I deliberately set the method to private because it's only for testing so it's not on the public interface

Reply:
I just had another look and decided I will change it, added a small `closer` interface that can be used in the test, WDYT?

Reply:
Nice -- this looks super clean. I like it.